### PR TITLE
Please enter the commit message for your changes. Lines starting

### PR DESCRIPTION
**Pull request by SvetaHBar**

### CODE REVIEW

**Comment by EdDev** [comment]
Please fix the commit message and then the poll request name.

File: src/cactus.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+#
+import time
+from datetime import date
+
+class Cactus(object):
+
+    def __init__(self, kind, watering):
+        self._kind = kind
+        self._watering = watering
+        self._is_succulent = True
```
Comment (EdDev) [inline src/cactus.py]: What is this suppose to represent? A kind/type?

File: src/cactus.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+# Refer to the README and COPYING files for full details of the license
+#
+import time
+from datetime import date
+
+class Cactus(object):
+
+    def __init__(self, kind, watering):
+        self._kind = kind
+        self._watering = watering
```
Comment (EdDev) [inline src/cactus.py]: Why "watering" is an inputted value? It does not fit as a default value?

File: src/cactus.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        self._is_succulent = True
+        self._is_plant_dry = True
+        self._is_plant_humid = True
```
Comment (EdDev) [inline src/cactus.py]: It is dry and humid at the same time?

File: src/cactus.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    def is_succulent(self):
+        return self._is_succulent
+
+    def need_to_water(self):
```
Comment (EdDev) [inline src/cactus.py]: One function should do the operation of water (to water).
A second function should ask if it is time to water.

When you perform the water operation, you stamp the current time (time.time() is good enough) on the cactus object. Then, when you ask if it is time to water it, it should get the current time, subtract the one saved on the object when last water operation occurred and calculate the diff.
Note that time() returns seconds, do the diff will be in seconds.

**Comment by SvetaHBar** [comment]
Please review

File: src/cactus.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        current_time = time.time()
+        print current_time
+        delta_time = current_time - self.watering_time
+        print delta_time
+        if delta_time <= 7777: #should be in seconds
+            return False
+        else:
+            return True
+    # more optimated way to write "if" :
+            #return delta_time > 7777
```
Comment (EdDev) [inline src/cactus.py]: You should use this version, not the noisy one.

File: src/cactus.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+    def watering(self):
+        self.watering_time = time.time()
+        print self.watering_time
+
+
+
+    def need_to_water(self):
+
+        current_time = time.time()
+        print current_time
```
Comment (EdDev) [inline src/cactus.py]: No prints in production code please.

File: src/cactus.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+# Refer to the README and COPYING files for full details of the license
+#
+import time
+import datetime
+
+class Cactus(object):
+
+    def __init__(self, kind):
+        self._kind = kind
+        #self._watering = watering
```
Comment (EdDev) [inline src/cactus.py]: Remove the leftovers please.

File: src/cactus.py [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+        self._is_succulent = True
+        self._is_plant_dry = True
+        self._is_plant_humid = False
+        self.watering_time = 0
+        self.potter = 0
+        self.is_need_sun = True
```
Comment (EdDev) [inline src/cactus.py]: Drop the 'is', it does not fit.

File: src/cactus.py [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+class Cactus(object):
+
+    def __init__(self, kind):
+        self._kind = kind
+        #self._watering = watering
+        self._is_succulent = True
+        self._is_plant_dry = True
+        self._is_plant_humid = False
+        self.watering_time = 0
```
Comment (EdDev) [inline src/cactus.py]: Needs to be private, including others below I guess.

File: src/cactus.py [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+            return False
+        else:
+            return True
+    # more optimated way to write "if" :
+            #return delta_time > 7777
+    #def is_plant_dry(self):
+        #return self._is_plant_dry
+
+    def is_need_large_potter(self):
+        current_potter = time.time()
```
Comment (EdDev) [inline src/cactus.py]: you mean current_time, right?

File: src/cactus.py [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+        self.potter = time.time()
+
+    def is_need_sun(self):
+        return self.is_need_sun
+
+    def move_to_shadow(self):#if cactus should not be on the sun , move them to shadow
+        self.is_need_sun = False
+
+
+def foo():
```
Comment (EdDev) [inline src/cactus.py]: Does not belong here, probably should be dropped.

**Pull request merged**